packages/flutter_tools/test/general.shard/commands/build_android_test.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:flutter_tools/src/android/android_sdk.dart';
+import 'package:flutter_tools/src/android/gradle.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/build_apk.dart';
+import 'package:flutter_tools/src/commands/build_appbundle.dart';
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/runner/flutter_command.dart';
+import 'package:mockito/mockito.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+
+class CurrentDirectoryFlutterProjectFactory extends FlutterProjectFactory {
```

Comment:
nit: move this class after the main() function, since it's less relevant than the actual tests.  Also, why do you need this class? You can mock the current directory.

Reply:
Moved.

I couldn't mock the current directory since this method requires a real `fs` object to have access to filesystem. Is there a way to mock only a single method of the filesystem object?